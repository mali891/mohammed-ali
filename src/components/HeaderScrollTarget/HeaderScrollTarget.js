import React from 'react'
import PropTypes from 'prop-types'

import { styles } from './HeaderScrollTarget.styles'

const HeaderScrollTarget = ({ scrollRef, ...other }) => {
  return <div className="c-header-scroll" css={styles} ref={scrollRef} {...other}></div>
}

HeaderScrollTarget.propTypes = {
  scrollRef: PropTypes.object.isRequired
}

export default HeaderScrollTarget
