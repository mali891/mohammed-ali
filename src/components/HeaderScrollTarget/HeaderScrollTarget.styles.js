import { css } from '@emotion/react'

export const styles = css`
  display: block;
  width: 100%;
  height: 1px;
`
