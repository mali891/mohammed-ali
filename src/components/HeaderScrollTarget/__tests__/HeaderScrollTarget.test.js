import React from 'react'
import { render } from '@testing-library/react'

import HeaderScrollTarget from '../HeaderScrollTarget'

describe('HeaderScrollTarget', () => {
  it('renders correctly', () => {
    const { asFragment } = render(<HeaderScrollTarget scrollRef={{ current: {} }} />)
    expect(asFragment()).toMatchSnapshot()
  })
})
