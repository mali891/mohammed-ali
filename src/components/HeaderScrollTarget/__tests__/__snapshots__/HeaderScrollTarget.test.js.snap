// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderScrollTarget renders correctly 1`] = `
<DocumentFragment>
  <div
    class="c-header-scroll"
    css="[object Object]"
  />
</DocumentFragment>
`;
